Add release notes for Kyma 2.13 (#974)
kyma-project/website: content/blog-posts/2023-04-14-release-notes-2.13/index.md

---
title: "Kyma 2.13"
author:
  name: "Wojciech Soltys, RM @Kyma, and Iwona Langer, Technical Writer @Kyma"
tags:
  - release-notes 
type: release 
releaseTag: "2.13.0"
redirectFrom:
  - "/blog/release-notes-213"
---

Like the sea that relentlessly sends waves softly shaping the shore, we also incessantly keep sending new releases to you and hope to improve your Kyma experience gently but consistently. Here we come, with version 2.13, bringing a collection of updates, fixes, and improvements that will make you love Kyma even more! Read on to find out what we have prepared for you.

<!-- overview -->

See the overview of all changes in this release:

- [API Gateway](#api-gateway) - local host `trust` authentication restricted in favor of MD5 authentication

> - [Observability](#observability) -  Kube-state-metrics update

Reviewer: ```suggestion
- [Observability](#observability) -  updated kube-state-metrics
```
Capitalization as spotted by Natalia; reverse word order to be consistent with the following line

- [Telemetry](#telemetry) - improved caching of Kubernetes resources; improved validation of LogPipelines; auto-generated CRD reference documentation; improved setup resiliency; OTel Collector update
- [Serverless](#serverless) - `serverless.kyma-project.io/v1alpha1` API version and dependencies to cluster-essentials removed
- [Security](#security) - Istio upgraded to 1.17.1

## API Gateway  

> From now, the Ory Hydra Postgres database refuses localhost connections without a password, as we have introduced MD5 authentication. Check [here](https://github.com/kyma-project/kyma/pull/17138) for more details.

Reviewer 2: ```suggestion
From now, the Ory Hydra Postgres database refuses localhost connections without a password, as we have introduced MD5 authentication. Check [this PR](https://github.com/kyma-project/kyma/pull/17138) for more details.
```
Wouldn't it be better to use more meaningful words instead or "this", "that", "here"? What do you think?

Reviewer: YESSSS

Also, we have updated API Gateway to version 1.5.0 and thus introduced a default request time for the exposed workload.

## Observability  
### Monitoring  

>   With the 2.13 version of Kyma, we've [updated Kube-state-metrics to version 2.8.1](https://github.com/kyma-project/kyma/pull/17058).

Reviewer: I agree! Thanks for spotting this!

## Telemetry  
### Manager  

This release brings improvement in the way Telemetry Manager caches Kubernetes resources. As a result, the manager has a reduced memory footprint and requires fewer access permissions to Kubernetes resources. For more information on the changes, see [this issue](https://github.com/kyma-project/kyma/issues/17001). 

We have also [improved the validation of LogPipelines by the webhook](https://github.com/kyma-project/kyma/issues/15480), which allows for earlier feedback reception. 

From now, the CRD reference documentation is always up-to-date as it's [automatically generated from the CRD description](https://github.com/kyma-project/kyma/issues/15663).

> We have improved the resiliency of the setup. Consequently, Fluent Bit DaemonSet is automatically recovered in case of unattended deletion. Check [here](https://github.com/kyma-project/telemetry-manager/pull/99) for more details.

Reviewer 2: ```suggestion
We have improved the resiliency of the setup. Consequently, Fluent Bit DaemonSet is automatically recovered in case of unattended deletion. Check [this PR](https://github.com/kyma-project/telemetry-manager/pull/99) for more details.
```

Similar suggestion to my previous one.

### Tracing  

> With Kyma 2.13, we've [updated OTel Collector to version 0.74.0](https://github.com/kyma-project/kyma/pull/17058).

Reviewer: @a-thaler please confirm.

@nataliasitko I'm beyond impressed by your thoroughness.

### Logging  

> With this release, [Telemetry production profiles have been removed](https://github.com/kyma-project/kyma/issues/16853), and production settings are the new default.

Reviewer 3: Its removed but also now the default?
This means its not a dedicated profile anymore?

Reviewer 4: How about this: 
`With this release, Telemetry production profiles have been removed, and production settings are the new default managed by dedicated module managers.`?
Does it make more sense to you, @k15r? Is it correct, @a-thaler?

Reviewer 4: > What about this: The profiles for the telemetry component got consolidated and no specialized settings for profiles are in use anymore. The new default are the settings from the previous production profile.

Thank you.

We have [updated Fluent Bit to version 2.0.10](https://github.com/kyma-project/kyma/pull/17109) and also, [improved its retry handling](https://github.com/kyma-project/kyma/issues/17113) by increasing the Fluent Bit retry limit.

## Serverless  
###  `serverless.kyma-project.io/v1alpha1` API version removed  

Following the depreciation of the `serverless.kyma-project.io/v1alpha1` API version with [Kyma 2.6](https://kyma-project.io/blog/2022/8/25/release-notes-26/#serverless), we've finally removed it in this release. The conversion webhook responsible for converting v1alpha1 to v1alpha2 during the grace period has also been removed.

### Removed dependencies to cluster-essentials

> From now, the Serverless module brings all its belongings within its Helm Chart and becomes independent of the cluster-essentials module. It is a step towards the modularization of Kyma, in which independent modules are the basic concept.

Reviewer: ```suggestion
From now on, the Serverless module brings all its belongings within its Helm Chart and becomes independent of the cluster-essentials module. It is a step towards the modularization of Kyma, in which independent modules are the basic concept.
```

---
title: "Kyma 2.13"
author:
  name: "Wojciech Soltys, RM @Kyma, and Iwona Langer, Technical Writer @Kyma"
tags:
  - release-notes 
type: release 
releaseTag: "2.13.0"
redirectFrom:
  - "/blog/release-notes-213"
---

#TODO

<!-- overview -->

See the overview of all changes in this release:

- [API Gateway](#api-gateway) - #TODO
- [Observability](#observability) -  #TODO 
- [Security](#security) - #TODO
- [Serverless](#serverless) - #TODO
- [Telemetry](#telemetry) - #TODO

## API Gateway
Ory Hydra Postgres DB do not allow localhost connections without password anymore https://github.com/kyma-project/kyma/pull/17138

With this Kyma release API Gateway was updated to version 1.5.0, which introduced default request time for exposed workload.

> ## Observability

Reviewer: I suggest we put Observability and Telemetry right after one another; not so far away like they are now.

Author: fixed

Reviewer: thank you!

### Monitoring

- Update of components:
    - Kube-state-metrics 2.8.1 https://github.com/kyma-project/kyma/pull/17058

## Telemetry

### Manager

> - The way how the Telemetry manager caches Kubernetes resources has been improved, which reduces the memory footprint. Also, the manager now needs fewer roles (https://github.com/kyma-project/kyma/issues/17001).

Reviewer: ```suggestion
- The way how the Telemetry manager caches Kubernetes resources has been improved, so it needs fewer permissions for access to Kubernetes resources. Also, this reduces the memory footprint (https://github.com/kyma-project/kyma/issues/17001).
```

> - Improved validation of LogPipelines by the webhook, so that you get feedback earlier (https://github.com/kyma-project/kyma/issues/15480).

Reviewer: ```suggestion
- Improved validation of LogPipelines by the webhook (https://github.com/kyma-project/kyma/issues/15480).
```

- The CRD reference documentation is generated from the CRD description, so it's always up to date (https://github.com/kyma-project/kyma/issues/15663).

> - Improved resiliency if the setup by recovering an unattended deletion of fluent-bit  (https://github.com/kyma-project/telemetry-manager/pull/99)

Reviewer: ```suggestion
- Improved resiliency of the setup: If Fluent Bit was deleted, it’s automatically recovered (https://github.com/kyma-project/telemetry-manager/pull/99).
```

### Tracing
- Update of components:

> - Otel-collector 0.74.0 https://github.com/kyma-project/kyma/pull/17058

Reviewer: ```suggestion
  - Otel Collector 0.74.0 https://github.com/kyma-project/kyma/pull/17058
```

### Logging

> - Consolidated profiles https://github.com/kyma-project/kyma/issues/16853

Reviewer: ```suggestion
- Profiles have been removed and production settings are the new default (https://github.com/kyma-project/kyma/issues/16853).
```

- Update of components:

>   - Fluentbit 2.0.10 https://github.com/kyma-project/kyma/pull/17109

Reviewer: ```suggestion
  - Fluent Bit 2.0.10 (https://github.com/kyma-project/kyma/pull/17109)
```

>   - Improved retry handling (https://github.com/kyma-project/kyma/issues/17113)

Reviewer: one level higher
```suggestion
```

Serverless module brings all it's belongings within its helm chart and becomes independant on  cluster-essentials module.
This is a step towards kyma modularisation where independant modules are the basic concept.

### Maintanance

We have introduced multiple security patches and fixed a [bug](https://github.com/kyma-project/kyma/issues/17147) related to premature Function CR creation

## Telemetry

### Manager

> - The way how the manage is caching kubernetes resources got improved and with that it has a reduced the memory footprint and a reduced set of roles is required by the manager (https://github.com/kyma-project/kyma/issues/17001)

Reviewer: ```suggestion
- The way how the Telemetry manager caches Kubernetes resources has been improved, which reduces the memory footprint. Also, the manager now needs fewer roles (https://github.com/kyma-project/kyma/issues/17001).
```

> - Improved validation of LogPipelines by the webhook so that you get feedback more early (https://github.com/kyma-project/kyma/issues/15480)

Reviewer: ```suggestion
- Improved validation of LogPipelines by the webhook, so that you get feedback earlier (https://github.com/kyma-project/kyma/issues/15480)
```

> - The CRD reference documentation gets generated out of the CRD description and with that will not be outdated anymore (https://github.com/kyma-project/kyma/issues/15663)

Reviewer: ```suggestion
- The CRD reference documentation is generated from the CRD description, so it's always up to date (https://github.com/kyma-project/kyma/issues/15663).
```

> - Improved resiliency if the setup by recovering an unattended deletion of fluent-bit  (https://github.com/kyma-project/telemetry-manager/pull/99)

Reviewer: ```suggestion
- Improved resiliency of the setup: If Fluent Bit was deleted unintentionally, it’s automatically recovered (https://github.com/kyma-project/telemetry-manager/pull/99).
```
or maybe 
```suggestion
- Improved resiliency of the setup: If Fluent Bit was deleted unintentionally, it can be recovered (https://github.com/kyma-project/telemetry-manager/pull/99).
```

### Tracing
- Update of components:

> - Otel-collector 0.74.0 https://github.com/kyma-project/kyma/pull/17058

Reviewer: ```suggestion
  - Otel Collector 0.74.0 https://github.com/kyma-project/kyma/pull/17058
```
